tests/components/recorder/test_util.py

```diff
@@ -575,9 +576,12 @@ async def test_write_lock_db(hass, tmp_path):
     instance = hass.data[DATA_INSTANCE]
 
     def _drop_table():
+        # This will deadlock in tests with the MutexPool
         with instance.engine.connect() as connection:
             connection.execute(text("DROP TABLE events;"))
 
+    # This will issue a warning about being called from the wrong thread.
+    # I think the test should use the lock database API instead of calling this directly
```

Comment:
Comment added to explain why the test fails, to be removed before the PR is merged